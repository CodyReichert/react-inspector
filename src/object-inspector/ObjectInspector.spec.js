import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ObjectInspector from './ObjectInspector';

const TestRenderer = new ShallowRenderer();

describe('ObjectInspector', () => {
  it('should render', () => {
    const tree = TestRenderer.render(<ObjectInspector theme="testvalue" />);

    expect(tree).toMatchSnapshot();    
    
    expect(tree.type).toBeInstanceOf(Function);    
    expect(tree.props.theme).toEqual('testvalue');
  });

  it('passes `nodeRenderer` prop to <TreeView/>', () => {
    // Test that a custom `nodeRenderer` props is passed to <TreeView/>
    const nodeRenderer = () => <span>unit test</span>;
    const tree = TestRenderer.render(<ObjectInspector nodeRenderer={nodeRenderer} />);

    expect(tree).toMatchSnapshot();    
    
    expect(tree.props.children.type).toBeInstanceOf(Function);    
    expect(tree.props.children.props.nodeRenderer).toEqual(nodeRenderer);
  });
});
