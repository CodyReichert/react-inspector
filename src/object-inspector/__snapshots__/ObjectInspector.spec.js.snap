// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectInspector passes \`nodeRenderer\` prop to <TreeView/> 1`] = `
<ThemeProvider
  theme="chromeLight"
>
  <TreeView
    dataIterator={[Function]}
    name={undefined}
    nodeRenderer={[Function]}
    theme="chromeLight"
  />
</ThemeProvider>
`;

exports[`ObjectInspector should render 1`] = `
<ThemeProvider
  theme="testvalue"
>
  <TreeView
    dataIterator={[Function]}
    name={undefined}
    nodeRenderer={[Function]}
    theme="testvalue"
  />
</ThemeProvider>
`;
